### DISCUSSION TITLE
Graph Pooling without batch

### DISCUSSION THREAD

**Discussion by dgm2** [pyg-team/pytorch_geometric#7201] [Q&A]
I have a batch of graphs features of dimension `bs, 18, 18` and its adjacency matrix also `bs,18,18`.
what is a way to pool this into `bs,1` e.g. each tensor pooled in single value ?
Any better way just getting the mean e.g. with some nn layer?
I looked into `pyg` pooling layers but they require some batch tensor that I don't have.

many thanks

**Comment by wsad1** [comment]
You have two options.
1. Use [Dense Pooling Layers](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#dense-pooling-layers)
2. Convert your adjacency matrix to sparse version using [dense_to_sparse](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.dense_to_sparse), and reshape your feature vector 'x.view(bs*18, 18)'. Then use any [pooling layer](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#pooling-layers)

**Reply by dgm2** [reply]
thanks, for option 1, how do I get the parameter `s (Tensor): Assignment tensor`.
The issue I am finding, is there is this other parameter about "clusters" required in pooling?
I only have `x` and `adj`

**Reply by rusty1s** [reply]
The `s` tensor is usually learned, e.g., through an MLP on the node embeddings.

**Reply by dgm2** [reply]
thanks for explaining.
what should be the dimension of `s` in this case? are there minimal examples of pooling ?

**Reply by wsad1** [reply]
you can learn `s` using a GNN like done [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/proteins_diff_pool.py).
